pkg/controller/clusters/management_test.go

```diff
@@ -0,0 +1,115 @@
+package clusters_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/clusters"
+)
+
+func TestFetchManagementEksaClusterSuccess(t *testing.T) {
+	tt := newFetchManagementTest(t)
+
+	objs := []runtime.Object{tt.cluster, tt.managementCluster}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err).To(BeNil())
+	tt.Expect(managementCluster).To(Equal(tt.managementCluster))
+}
+
+func TestFetchManagementEksaClusterDifferentNamespaceSuccess(t *testing.T) {
+	tt := newFetchManagementTest(t)
+	tt.managementCluster.Namespace = "different-namespace"
+
+	objs := []runtime.Object{tt.cluster, tt.managementCluster}
+	cb := fakeClientBuilder()
+	cl := cb.WithRuntimeObjects(objs...).Build()
+
+	managementCluster, err := clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)
+	tt.Expect(err).To(BeNil())
+	tt.Expect(managementCluster).To(Equal(tt.managementCluster))
```

Comment:
nit: a trick I learnt recently
```suggestion
	tt.Expect(clusters.FetchManagementEksaCluster(context.Background(), cl, tt.cluster)).To(Equal(tt.managementCluster))
```

This will take care of checking the error value to nil afaik

Reply:
I thought I tried that and saw that it complained about two return values, but maybe it was something else